src/ontodia/data/rdf/rdfDataProvider.ts

```diff
@@ -0,0 +1,619 @@
+                    );
+                });
+            }).catch(error => {
+                console.warn(error);
+                return null;
+            }),
+        );
+
+        return Promise.all(queries).then((fetchedModels) => {
+            return fetchedModels.filter(lt => lt);
+        });
+    }
+
+    linkTypes(): Promise<LinkType[]> {
+        const linkTypes: LinkType[] = [];
+        const rdfLinks = this.rdfStorage.match(
+            undefined,
+            PREFIX_FACTORIES.RDF('type'),
+            PREFIX_FACTORIES.RDF('Property'),
+        );
+        const owlLinks = this.rdfStorage.match(
+            undefined,
+            PREFIX_FACTORIES.RDF('type'),
+            PREFIX_FACTORIES.OWL('ObjectProperty'),
+        );
+        return Promise.all([rdfLinks, owlLinks]).then(props => {
+            const links = props[0].toArray().concat(props[0].toArray());
+            return Promise.all(
+                links.map(l =>
+                    this.getLabels(l.subject.nominalValue).then(labels => {
+                        return {
+                            id: l.subject.nominalValue,
+                            label: { values: labels },
+                            count: this.rdfStorage.getTypeCount(l.subject.nominalValue),
+                        };
+                    }),
+                ),
+            );
+        });
+    }
+
+    elementInfo(params: { elementIds: string[] }): Promise<Dictionary<ElementModel>> {
+        const elementInfoResult: Dictionary<ElementModel> = {};
+
+        const queries = params.elementIds.map(
+            elementId => this.rdfStorage.checkElement(elementId).then(exists => {
+                return this.fetchIfNecessary(elementId, exists).then(() => {
+                    return this.getElementInfo(elementId);
+                });
+            }).catch(error => {
+                console.warn(error);
+                return null;
+            }),
+        );
+
+        return Promise.all(queries).then((fetchedModels) => {
+            for (const model of fetchedModels) {
+                if (model) {
+                    elementInfoResult[model.id] = model;
+                }
+            }
+            return elementInfoResult;
+        });
+    }
+
+    linksInfo(params: {
+        elementIds: string[];
+        linkTypeIds: string[];
+    }): Promise<LinkModel[]> {
+
+        const statementPromises: Promise<MatchStatement>[] = [];
+        for (const source of params.elementIds) {
+            for (const target of params.elementIds) {
+                statementPromises.push(Promise.all([
+                    this.rdfStorage.checkElement(source)
+                        .then(exists => this.fetchIfNecessary(source, exists)),
+                    this.rdfStorage.checkElement(target)
+                        .then(exists => this.fetchIfNecessary(target, exists)),
+                ]).then(() => (
+                    {subject: source, predicat: undefined, object: target}
```

Comment:
You misspelled `predicate` here. You could explicitly specify a return type for a lambda so compiler could catch this kind of errors.